services/supervisor/src/proto/service.proto

```diff
@@ -33,9 +33,11 @@ message DeployRequest {
 }
 
 // Confirmation that the artifact was successfully deployed by the validator.
-message DeployConfirmation {
+message DeployResult {
   // Artifact identifier.
-  exonum.runtime.ArtifactId artifact = 1;
+  DeployRequest request = 1;
+  // Result of deployment.
+  bool success = 2;
```

Comment:
I'm not sure whether the supervisor should make nodes agree on the error, but at least recording the error on the blockchain would be useful for auditing purposes.